dfs/registry/index.go

```diff
@@ -38,6 +39,7 @@ var _ = RegistryIndex(&RegistryIndexClient{})
 type facade interface {
 	GetRegistryImage(ctx datastore.Context, image string) (*registry.Image, error)
 	SetRegistryImage(ctx datastore.Context, rImage *registry.Image) error
+	SetRegistryImageAfterCommit(ctx datastore.Context, rImage *registry.Image) error
```

Comment:
Why do you need a SetAfterCommit?  SetRegistryImage indiscriminantly writes a value to the database